plugins/outputs/sematext/processors/rename.go

```diff
@@ -0,0 +1,149 @@
+		"mongodb.mapped_megabytes":        "memory.mapped",
+		"mongodb.inserts":                 "ops.insert",
+		"mongodb.queries":                 "ops.query",
+		"mongodb.updates":                 "ops.update",
+		"mongodb.getmores":                "ops.getmore",
+		"mongodb.commands":                "ops.command",
+		"mongodb.repl_inserts":            "replica.ops.insert",
+		"mongodb.repl_queries":            "replica.ops.query",
+		"mongodb.repl_updates":            "replica.ops.update",
+		"mongodb.repl_deletes":            "replica.ops.delete",
+		"mongodb.repl_getmores":           "replica.ops.getmore",
+		"mongodb.repl_commands":           "replica.ops.command",
+		"mongodb.count_command_failed":    "commands.failed",
+		"mongodb.count_command_total":     "commands.total",
+		"mongodb_db_stats.data_size":      "database.data.size",
+		"mongodb_db_stats.storage_size":   "database.storage.size",
+		"mongodb_db_stats.index_size":     "database.index.size",
+		"mongodb_db_stats.collections":    "database.collections",
+		"mongodb_db_stats.objects":        "database.objects",
+		"mongodb_db_stats.avg_obj_size":   "database.avg_obj_size",
+		"mongodb_db_stats.indexes":        "database.indexes",
+		"mongodb_db_stats.num_extents":    "database.num_extents",
+		"mongodb_db_stats.ok":             "database.ok",
+		"mongo.connections_current":       "network.connections",
+		"mongo.connections_total_created": "network.connections.total",
+		"mongo.net_in_bytes":              "network.transfer.rx.rate",
+		"mongo.net_out_bytes":             "network.transfer.tx.rate",
+		// mongodb_col_stats -> these appear like they map to the same thing, but "from" side is actually
+		// "name.metricName" and "to" side is just the new "metricName"
+		"mongodb_col_stats.avg_obj_size":     "mongodb_col_stats.avg_obj_size",
+		"mongodb_col_stats.count":            "mongodb_col_stats.count",
+		"mongodb_col_stats.ok":               "mongodb_col_stats.ok",
+		"mongodb_col_stats.size":             "mongodb_col_stats.size",
+		"mongodb_col_stats.storage_size":     "mongodb_col_stats.storage_size",
+		"mongodb_col_stats.total_index_size": "mongodb_col_stats.total_index_size",
+		// mongodb_shard_stats -> same logic as for mongodb_col_stats
+		"mongodb_shard_stats.in_use":     "mongodb_shard_stats.in_use",
+		"mongodb_shard_stats.available":  "mongodb_shard_stats.available",
+		"mongodb_shard_stats.created":    "mongodb_shard_stats.created",
+		"mongodb_shard_stats.refreshing": "mongodb_shard_stats.refreshing",
+	}
+)
+
+// Rename processor renames the measurement (metric) names
+// to match the existing metric names sent by Node.js agents
+type Rename struct{}
+
+// NewRename builds a new rename processor.
+func NewRename() BatchProcessor { return &Rename{} }
+
+// Process performs a lookup in the local maps of metric/field names
+// and replaces the metric name with the new name.
+func (r *Rename) Process(points []telegraf.Metric) ([]telegraf.Metric, error) {
```

Comment:
I think this processor is totally flawed. You can easily rename metrics in telegraf using the `name_override`, `name_prefix` etc functions (see [config options](https://github.com/influxdata/telegraf/blob/master/docs/CONFIGURATION.md#input-plugins)). Therefore you might miss those metrics. Is there any reason why you need to "detect" the type of metric?